app/src/components/GrantDetailsRow.vue

```diff
@@ -86,6 +88,7 @@ export default defineComponent({
   components: { CartIcon, LogoPtrImage },
   props: {
     grant: { type: Object as PropType<Grant>, required: true },
+    idList: { type: Array as PropType<number[]>, required: true },
```

Comment:
Moving the fetching and storage of the whitelist to the `data` store feels like a much better architecture here.  As is, it seems we re-fetch the whitelist every time a different grant's detail page is loaded, which is wasteful and bad UX. We should fix that here, instead of coupling these components further

Reply:
Yeah I think this is `tech-debt` which I thought got resolved in a different PR, right now this is a hotfix considering the issue is high priority. I'm now figuring out how to quickly make proper changes here.

Reply:
Just add a `whitelistUrl` ref to the data store, return it as a computed property at the bottom of that file, and fetch the whitelist during `init` or `rawInit` (not sure where exactly is best offhand, cc @gdixon), then import it into the components

Reply:
sounds good - I was mulling between what you're suggesting but also if it makes sense to get it from `grants.ts` since  we are fetching it here already https://github.com/dcgtc/dgrants/blob/main/app/src/utils/data/grants.ts#L241

Reply:
seems like we have that in data store too, so I should be good - it probably never got changed in the `GrantRegistryGrantDetail.vue` file

Reply:
Ah so then you just need to use the `approvedGrantsPk` that's returned from `data.ts`, which is where you're linked function is used. Even easier then 🙂